src/tir/ir/expr.cc

```diff
@@ -1056,12 +1058,28 @@ TVM_STATIC_IR_FUNCTOR(ReprPrinter, vtable)
     .set_dispatch<AnyNode>([](const ObjectRef& node, ReprPrinter* p) { p->stream << "?"; });
 
 // BufferLoad
+void BufferLoadNode::LegalizeDtype() {
+  int index_lanes = 1;
```

Comment:
It is over generalization to allow vector indices in all dimensions. Consider restricting the vector indices to only in the last dimension for now `indices[-1]`.

Reply:
Definitely agreed that this is an overgeneralization, and should be restricted further.  One concern with restricting it that much is that it might not handle the case a vector index in an earlier index, which is then moved to the last index by a subsequent layout transformation.

What if we restrict it to have at most one vector index, but allow that index to occur in any location?  That would avoid questions on how multiple vector indices should be flattened (e.g. I could see `[ramp(i,1,4),ramp(j,1,4)]` on a `[4,4]` buffer being reasonable flattened to `[ramp(4*i + j, 1, 4)]` or to `[ramp(ramp(j,1,4), 1,4)]`, or to `[ramp(i, 1, 16)]`, depending on how the two indices interact), but would still allow for the transformed-to-last-index case.

Reply:
I agree that could be an useful generalization. 

The main rational of the suggestion is that we might want to start small(that matches the original behavior) then generally analyze the effect and generalize to cases(like at most one vector indices) after thinking about all the consequences. This is mainly to avoid possibly bugs on unsupported cases that we do not yet anticipated

Assuming right now we still restrict vectorize to happen after flattening, the last dimension restriction could serve the current purose, then we can discuss for more generlaization.